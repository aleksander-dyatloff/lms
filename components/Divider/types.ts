namespace DividerComponent {
  export interface Props {
    dir?: 'horizontal' | 'vertical'
    className?: string
  }
}

export default DividerComponent
