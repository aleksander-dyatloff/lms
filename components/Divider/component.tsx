import combineClasses from '@utils/combineClasses'
import { FC } from 'react'

import Wrapper from './styles'
import DividerComponent from './types'

const Divider: FC<DividerComponent.Props> = ({
  dir = 'horizontal',
  color,
  className,
  ...restProps
}) => {
  const wrapperClasses = combineClasses(dir, className)

  return (
    <Wrapper
      className={wrapperClasses}
      {...restProps}
    />
  )
}

export default Divider
