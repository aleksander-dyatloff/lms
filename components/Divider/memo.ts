import { memo } from 'react'

import Divider from './component'

export default memo(Divider)
