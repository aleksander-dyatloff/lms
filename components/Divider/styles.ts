import Element from '@components/Element/styles'
import styled from 'styled-components'

const Wrapper = styled(Element)`
  background-color: ${({
    color, theme,
  }) => color ?? theme.palette.primary.contrast};

  &.vertical {
    width: 2px;
    height: 100px;
  }

  &.horizontal {
    width: 100%;
    height: 2px;
  }
`

export default Wrapper
