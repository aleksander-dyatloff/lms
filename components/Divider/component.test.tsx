import 'jest-styled-components'

import { render } from '@testing-library/react'
import { Profiler } from 'react'

import Divider from './component'

describe('Divider component', () => {
  test('should match to snapshot', () => {
    const { container } = render(<Divider />)

    expect(container).toMatchSnapshot()
  })

  test('first render duration should match to specification', () => {
    render(
      <Profiler
        id='Divider'
        onRender={(_, __, renderDuration) => {
          expect(renderDuration).toBeLessThanOrEqual(1.5)
        }}
      >
        <Divider />
      </Profiler>,
    )
  })
})
