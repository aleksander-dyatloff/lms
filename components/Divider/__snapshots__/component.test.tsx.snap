// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component should match to snapshot 1`] = `
.c0 {
  padding: 0;
  margin: 0;
  font: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  list-style: none;
  color: inherit;
  background-color: inherit;
  border: none;
  border-radius: 0;
  width: 100%;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  cursor: inherit;
  outline: none;
  caret-color: inherit;
}

.c1 {
  background-color: gray;
}

.c1.vertical {
  width: 1px;
  height: 100px;
}

.c1.horizontal {
  width: 100%;
  height: 1px;
}

<div>
  <div
    class="c0 c1  horizontal"
  />
</div>
`;
